build.gradle

```diff
@@ -119,6 +114,9 @@ allprojects {
             libs.nettyHandler,
             libs.nettyTransportNativeEpoll
           )
+          dependencySubstitution {
+            substitute module("log4j:log4j:1.2.17") because "we use a custom version with security patches" with module("io.confluent:confluent-log4j:1.2.17-cp2")
+          }
         }
```

Comment:
@rajinisivaram  My immediate reaction is that this `dependencySubstitution` block is conditional on `name != "zinc"`, which seems to be not what we'd like. We'd like this module substitution regardless. But then, I am not too familiar with zinc (and scala ecosystem in general). 
Is there a quick way for me to build in a way where `name == "zinc"`? I'll be happy to test.

Reply:
@niteshmor Thanks for the review. Zinc is the scala incremental compiler, I don't think `dependencySubstitution` needs to apply to that. @ijuma Can you confirm please?

Reply:
@niteshmor any build for a scala project will include a `zinc` configuration. The zinc configuration is only used during compilation, so I think what we have here is fine.

Reply:
If that's the case, then this is fine. Thanks Ismael.
@rajinisivaram, you have my +1 for the PR as far `build.gradle` is concerned.